crates/client/l1-messages/src/error.rs

```diff
@@ -0,0 +1,37 @@
+use mp_felt::Felt252WrapperError;
+
+#[derive(thiserror::Error, Debug, Eq, PartialEq)]
+pub enum L1MessagesConfigError {
+    #[error("File with L1 Messages Worker config not found: `{0}`")]
+    FileNotFound(String),
+    #[error("Failed to deserialize L1 Messages Worker Config from config file: `{0}`")]
+    InvalidFile(String),
+}
+
+#[derive(thiserror::Error, Debug, Eq, PartialEq)]
+pub enum L1MessagesWorkerError {
+    #[error("Failed to initialize L1 Messages Worker based on provided Config")]
+    ConfigError,
+    #[error("Failed to convert transaction via Runtime API")]
+    ConvertTransactionRuntimeApiError,
+    #[error("Madara Messaging DB Error: `{0}`")]
+    DatabaseError(String),
```

Comment:
keep it a error rather than a String

Reply:
Fair point, but in this case errors in most of `db` implementations are `String` type.

Reply:
This is a bad design. Try changing those method signatures so we can use the actual errors